ruby/app/controllers/articles.rb

```diff
@@ -1,43 +1,37 @@
 class ArticleController
   def create_article(article)
-    article_exists = Article.where(tilte: 'title')
+    article_not_exists = ! (Article.where(:title => article['title']).empty?)
 
-    return { ok: false, msg: 'Article with given title already exists' } if article_exists
+    return { ok: false, msg: 'Article with given title already exists' } unless article_not_exists
 
-    new_article = Article.new
-    new_article.set_fields({ title: article['title'], content: article['content'], created_at: Time.now },
-                           %i[title content created_at])
-    new_article.save_changes
+    new_article = Article.new(:title => article['title'], :content => article['content'], :created_at => Time.now)
+    new_article.save
 
     { ok: false, obj: article }
   rescue StandardError
     { ok: false }
   end
 
   def update_article(id, new_data)
+
     article = Article.where(id: id).first
 
     return { ok: false, msg: 'Article could not be found' } unless article.nil?
 
-    article.set_fields({ title: new_data['title'], content: new_data['content'] }, %i[title content])
+    article.title = new_data['title']
+    article.content = new_data['content']
     article.save_changes
```

Comment:
This is a new bug, because ActiveRecord has no `save_changes` method. It is just `save`. Therefore I removed the other bug in line 25.